shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/SchemaBuilder.java

```diff
@@ -84,43 +89,45 @@
     
     private static Map<String, TableMetaData> buildActualTableMetaDataMap(final SchemaBuilderMaterials materials) throws SQLException {
         Map<String, TableMetaData> result = new HashMap<>(materials.getRules().size(), 1);
-        appendRemainTables(materials, result);
-        for (ShardingSphereRule rule : materials.getRules()) {
-            if (rule instanceof TableContainedRule) {
-                for (String table : ((TableContainedRule) rule).getTables()) {
-                    if (!result.containsKey(table)) {
-                        TableMetaDataBuilder.load(table, materials).map(optional -> result.put(table, optional));
+        Collection<String> ruleLogicTables = materials.getRules().stream()
+                .filter(rule -> rule instanceof TableContainedRule)
+                .flatMap(rule -> ((TableContainedRule) rule).getTables().stream())
+                .collect(Collectors.toSet());
+        appendRemainTables(materials, result, ruleLogicTables);
```

Comment:
@Beyondeclipse It is better to adjust the execution order of the code instead of adding the `toAppendLoadTables` parameter.

Reply:
@Beyondeclipse If you adjust the order of execution, the result parameter will contain the rule table.

Reply:
> @Beyondeclipse If you adjust the order of execution, the result parameter will contain the rule table.

Can you give more explanation ?